### PR TITLE
coref_resolved added

### PR DESCRIPTION
**Pull request by linguist89**
I have now added the coref_resolved attribute at the Doc level from a clone of the main branch, so all other issues should be solved.

### CODE REVIEW

File: src/conspiracies/coref/CoreferenceComponent.py [hunk 1]
```diff
@@ -53,10 +53,12 @@ def __init__(
         self.model = CoreferenceModel(model_path=model_path, device=device)
 
         # Register custom extension on the Doc and Span
+        if not Doc.has_extension("resolved_coref"):
+            Doc.set_extension("resolved_coref", default=None)
         if not Doc.has_extension("coref_clusters"):
             Doc.set_extension("coref_clusters", default=[])
-        if not Span.has_extension("coref_clusters"):
-            Span.set_extension("coref_clusters", default=[])
+        if not Span.has_extension("coref_cluster"):
```
Comment (KennethEnevoldsen) [inline src/conspiracies/coref/CoreferenceComponent.py:60]: Why change it from coref_clusters? Seems like it is nice that is consistent with the Doc level?

File: src/conspiracies/coref/CoreferenceComponent.py [hunk 2]
```diff
@@ -83,8 +85,18 @@ def set_annotations(self, docs: Iterable[Doc], model_output) -> None:
...
+                            
+            # Add the resolved coreference doc
+            resolved = list(tok.text_with_ws for tok in doc)
+            for i, cluster in doc._.coref_clusters:
+                for coref in cluster:
+                    if coref != coref._.antecedent:
+                        resolved[coref.start] = coref._.antecedent.text + doc[coref.end-1].whitespace_
+                        for i in range(coref.start+1, coref.end):
+                            resolved[i] = ""
+            doc._.resolved_coref = ''.join(resolved)
```
Comment (KennethEnevoldsen) [inline src/conspiracies/coref/CoreferenceComponent.py:99]: Add this an extension to the Doc. No reason to compute this for all docs if not needed.

**Review by KennethEnevoldsen — CHANGES_REQUESTED** [review]
Looks good, but still a few things missing:
- [ ] It is missing a test convincing me that it works as intended.
- [x] We would ideally want to do this on a Span level as well.
- [ ] The code needs to be formatted to black, this can be done using:
```
# in the repository
pip install -r requirements.txt
# autoformat using black
black . 
```

File: src/conspiracies/coref/CoreferenceComponent.py (hunk 1 above)
Comment (linguist89) [reply src/conspiracies/coref/CoreferenceComponent.py]: Done

File: src/conspiracies/coref/CoreferenceComponent.py (hunk 2 above)
Comment (linguist89) [reply src/conspiracies/coref/CoreferenceComponent.py]: Done

File: tests/test_coref_component.py [hunk 3]
```diff
@@ -19,6 +19,9 @@ def test_coref_component(nlp_da):
     doc = nlp_da(text)
     # test attributes is set a
     assert isinstance(doc._.coref_clusters, list)
+    assert isinstance(doc._.resolve_coref, str)
```
Comment (KennethEnevoldsen) [inline tests/test_coref_component.py:22]: I would like to see that this actually resolved the coref as intended. I.e.

doc._.resolve_coref == "the desired string"

As there might naturally be errors in the coref res. it is perfectly valid to set the coref clusters manually as we are testing the functions here (not the model).

**Review by KennethEnevoldsen — CHANGES_REQUESTED** [review]

**Comment by KennethEnevoldsen** [comment]
Hmm, feel like these are still unresolved:
- [ ] It is missing a test convincing me that it works as intended (see comment).
- [ ] Add the coref_resolved as a Span extension as well.

**Review by KennethEnevoldsen — CHANGES_REQUESTED** [review]
see unresolved comment above. (this is just because I get pinged for not completing the review)

**Review by KennethEnevoldsen — APPROVED** [review]
Looks good will merge it in

**Pull request merged**